sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/ScriptTransformation.scala

```diff
@@ -25,9 +25,24 @@ import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
  * @param input the set of expression that should be passed to the script.
  * @param script the command that should be executed.
  * @param output the attributes that are produced by the script.
+ * @param ioschema the input and output schema applied in the execution of the script.
  */
 case class ScriptTransformation(
     input: Seq[Expression],
     script: String,
     output: Seq[Attribute],
-    child: LogicalPlan) extends UnaryNode
+    child: LogicalPlan,
+    ioschema: ScriptInputOutputSchema) extends UnaryNode
+
+/**
+ * The wrapper class of input and output schema properties for transforming with script.
+ *
+ */
+case class ScriptInputOutputSchema(
```

Comment:
Probably move the `ScriptInputOutputSchema` into `hive` package?  The `SerDe`, `RowFormat` are the concept of Hive. And the `ScriptTransformation` probably can be defined like:

```
case class ScriptTransformation[T](
     input: Seq[Expression],
     script: String,
     output: Seq[Attribute],
     child: LogicalPlan,
     ioschema: Option[T]) extends UnaryNode
```